src/MovieDetails/MoviePage.js

```diff
@@ -44,22 +45,34 @@ class MoviePage extends Component {
       })
   }
 
+  formatBudgetAndRevenue(x) {
+    if (x === 0) {
+      return 'Not Available'
+    } else {
+      let numWithCommas = x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")
+      return `$${numWithCommas}`
+    }
+  }
+
   render() {
+    let budget = this.formatBudgetAndRevenue(this.state.budget)
```

Comment:
I wonder if we should call this method after the component mounts, rather than in the render! So, we update the state of the MoviePage with 'Not Available' or '$500,000'